### PR TITLE
103: sharing implemented

### PR DESCRIPTION
**Pull request by yegor256**

### CODE REVIEW

**Comment by karato** [comment]
Let me find a reviewer for this pull request, thanks for submitting it

**Comment by karato** [comment]
@caarlos0 do this review

File: src/main/java/com/nerodesk/om/mock/MkFriends.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    @Override
+    public boolean leader() throws IOException {
+        return true;
+    }
+
+    @Override
+    public Iterable<String> names() throws IOException {
+        assert this.dir != null;
+        assert this.user != null;
+        assert this.label != null;
```
Comment (caarlos0) [inline src/main/java/com/nerodesk/om/mock/MkFriends.java:89]: ins't it better to put those validations in the constructor? With `@NonNull` or something like that..

**Comment by caarlos0** [comment]
@yegor256 just one comment above.

**Comment by yegor256** [comment]
@caarlos0 they are there only to fool PMD. if I put them into constructor, PMD will complain again

**Comment by yegor256** [comment]
@rultor merge (there was only one comment and I addressed it in the answer)

**Comment by yegor256** [comment]
@rultor merge again

**Comment by yegor256** [comment]
@rultor try to merge again

**Comment by karato** [comment]
@caarlos0 paid, thanks, added **16 mins** to your account, payment `AP-92J82904HL5598115`, 6 hours and 50 mins was spent... you're getting extra minutes here (c=1)... +16 added to your rating, current score is: [+1369](http://www.netbout.com/b/35095?open=rating)

**Comment by karato** [comment]
@rultor deploy

**Pull request merged**